compiler/damlc/daml-compiler/src/DA/Daml/Compiler/Repl.hs

```diff
@@ -419,6 +412,64 @@ runRepl importPkgs opts replClient logger ideState = do
             Left err -> liftIO $ renderError dflags err
             Right () -> pure ()
 
+    mkReplOption
+        :: (DynFlags -> [String] -> ExceptT Error ReplM ())
+        -> [String] -> ReplM ()
+    mkReplOption option args = do
+        ReplState {lineNumber} <- State.get
+        dflags <- liftIO $
+            hsc_dflags . hscEnv <$>
+            runAction ideState (use_ GhcSession $ lineFilePath lineNumber)
+        r <- runExceptT $ option dflags args
+        case r of
+            Left err -> liftIO $ renderError dflags err
+            Right () -> pure ()
+    replOptions :: [(String, [String] -> ReplM ())]
+    replOptions =
+      [ ("help", mkReplOption optHelp)
+      , ("module", mkReplOption optModule)
+      ]
+    optHelp _dflags _args = liftIO $ T.putStrLn $ T.unlines
+      [ " Commands available from the prompt:"
+      , ""
+      , "   <statement>                 evaluate/run <statement>"
+      , "   :module [+/-] <mod> ...     add or remove the modules from the import list"
+      ]
+    optModule _dflags ("-" : modules) = do
+        ReplState {imports} <- lift State.get
+        -- TODO[AH] Use a more appropriate data structure to track imports.
+        let unknown =
+              [ removed
+              | removed <- map mkModuleName modules
+              , removed `notElem` map (unLoc . ideclName) imports
+              ]
+            newImports =
+              [ simpleImportDecl imported
+              | imported <- map (unLoc . ideclName) imports
+              , imported `notElem` map mkModuleName modules
+              ]
+        unless (null unknown) $
+            throwError $ UnknownModules unknown
```

Comment:
`UnknownModules` sounds slightly confusing. If a module is in a package that we know about I would expect it to be known. Similarly, I would expect an import for a module in a package we don’t know about to count as unknown.

Maybe something like `NotImportedModules`. Not a great name so if you have a better suggestion go for it.

Reply:
Agreed, `NotImportedModules` is better.